client/packages/common/src/ui/components/navigation/AppNavLink/AppNavLink.tsx

```diff
@@ -99,27 +109,46 @@ export const AppNavLink: FC<AppNavLinkProps> = props => {
     drawer.onClick();
   };
 
+  const showConfirmation = useConfirmationModal({
+    onConfirm: handleClick,
+    message: confirmMessage || t('messages.confirm-delete-generic'),
+    title: confirmTitle || t('heading.are-you-sure'),
+  });
+
   const CustomLink = React.useMemo(
     () =>
-      React.forwardRef<HTMLAnchorElement>((linkProps, ref) =>
-        !end && !!inactive ? (
-          <span
-            {...linkProps}
-            onClick={() => drawer.onExpand(to)}
-            data-testid={`${to}_hover`}
-          />
-        ) : (
-          <Link
-            {...linkProps}
-            ref={ref}
-            to={to}
-            role="link"
-            aria-label={text}
-            title={text}
-            onClick={handleClick}
-          />
-        )
-      ),
+      React.forwardRef<HTMLAnchorElement>((linkProps, ref) => {
```

Comment:
I had wondered if in stead of this conditional statement, I could add an entirely different component to handle a confirmation-required-navLink.